scripts/lint_site.sh

```diff
@@ -18,9 +18,21 @@ set -e
 
 FAILED=0
 
-if [[ "$#" -ne 0 ]]; then
+if [[ -n "${BRANCH}" ]]; then
+    files_changed="$(git diff --name-only --diff-filter=b "$(git merge-base HEAD "$BRANCH")")"
+    if (echo "${files_changed}" | grep -q /en/); then
+        LANGS+="en "
+    fi
+    if (echo "${files_changed}" | grep -q /zh/); then
+        LANGS+="zh "
+    fi
+    if (echo "${files_changed}" | grep -q /pt-br/); then
+        LANGS+="pt-br "
+    fi
+elif [[ "$#" -ne 0 ]]; then
     LANGS="$*"
 else
+    echo "if BRANCH is set to the source branch, ie. master, the linter will only check languages within the PR."
```

Comment:
What PR? This message will make no sense in a local run of the script. It also wouldn't be clear to a user what "source branch" is. The message would need to be much more precise and explain something like the following: `Note: when running this script in the context of a branch from some "source branch", you can set the the BRANCH environment variable to the "source branch", which will instruct the linter to only run over the language files that include some modified files relative to the source branch`. Even that will confuse people, and given that we expect people to use make instead of calling the script directly, I suggest no message at all.
```suggestion
```